### DISCUSSION TITLE
Debug tips

### DISCUSSION THREAD

**Discussion by ste4net** [Newex/PartialSourceGen#20] [Q&A]
Is there a way to debug the library's execution? In my web solution, I can't get it to work when I use it in a DLL library project, though it works when used from the main console project. When creating a new test solution, it also works from a DLL library, so there must be some conflict within my web solution, but I'm not sure what to check.
Thankyou

**Comment by Newex** [comment]
The library is essentially just creating extra source code (source generated files).

You can emit the source generated files to see if the files look correct.

```xml
<!-- file: project.csproj -->
<PropertyGroup>
    <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
    <CompilerGeneratedFilesOutputPath>Generated</CompilerGeneratedFilesOutputPath>
</PropertyGroup>
```

This will emit source generated files to the `Generated` folder.

Remember to clean up the solution `dotnet clean` so you don't end up with multiple files in different locations. Multiple definitions for the same thing can create conflict and errors.

My source for this: https://andrewlock.net/creating-a-source-generator-part-6-saving-source-generator-output-in-source-control/#controlling-the-output-location

### Side note
I have noticed that a minimal api uses source generated files when mapping api-endpoints.

The problem is source generated files in the same project cannot see each others source generated files.

Example:

A minimal api project that defines `[Partial]` and uses `app.MapGet("/", (PartialEntity entity) => { /* todo */ });` in `Program.cs`.

Both `[Partial] Entity` and `app.MapGet` create source generated files - but they cannot see each other.

So `MapGet` cannot create the proper source files.

How I avoid this is by declaring an empty partial file like so:

```csharp
// First actual entity
[Partial]
public record Entity { }

// Create an empty partial file so that minimal api endpoint can see the type.
public partial record PartialEntity { }
```

Other than that, I cannot really say what is wrong with your specific use case without more info. Of course there could be a bug in the source code `PartialSourceGen`.

**Reply by ste4net** [reply]
Unfortunatelly after few test it stops working and i am not able to understand why.
With EmitCompilerGeneratedFiles and CompilerGeneratedFilesOutputPath i can se the generated file but there are a lot of compilation error because compiler saw two identical classes during compilation so i can't investigate to understand the reason behind the missing generated files. 
Correct myself.
This can be done excluding generated files from compilation with 
	<ItemGroup>
		<Compile Remove="Generated\**\*.g.cs" />
		<None Include="Generated\**\*.g.cs" />
	</ItemGroup>

**Comment by ste4net** [comment]
Thank you for the response. After multiple attempts made over different days, it has now started generating the code. I practically didn’t do anything except remove and reinstall the package, try to connect it to all the projects in the solution, and other attempts. Now, the classes are being generated both in the boot project and in the secondary ones. Thank you.

![image](https://github.com/user-attachments/assets/66079b44-878d-42c2-be24-c1c208ecc1ba)

**Reply by ste4net** [reply]
I probably found the issue  
The error occour when the class  with [Partial] inherit from a base class using  [Timestamp] attribute (System.ComponentModel.DataAnnotations)
It does not occour if the same base class is in the same assembly of the one i want to be used as generation source
The phisical file is generated but not the code under the analyzer element...

Case one (not working)...
```
//...[another assembly
    public class SImpleBase
    {
        [Timestamp]
        public byte[] RowVersion { get; set; } = new byte[8];
         
    }
 //...[another assembly
    [Partial]
    public class Prova : SImpleBase
    {
        public string Surname { get; set; }
        public string Name { get; set; }
    }
```
![image](https://github.com/user-attachments/assets/203e0cd6-ac73-4951-949b-74f6f49ce1f9)

Case one (working)...

```
public class SImpleBase2
    {
        [Timestamp]
        public byte[] RowVersion { get; set; } = new byte[8];

    }
    [Partial]
    public class Prova : SImpleBase2
    {
        public string Surname { get; set; }
        public string Name { get; set; }
    }
```

![image](https://github.com/user-attachments/assets/15e36938-74ce-438e-9111-a468fb9ce154)
